src/backend/linux_raw/thread/syscalls.rs

```diff
@@ -278,3 +282,9 @@ unsafe fn futex_old(
         c_uint(val3)
     ))
 }
+
+#[cfg(any(target_os = "android", target_os = "linux"))]
+#[inline]
+pub(crate) fn setns(fd: BorrowedFd, nstype: c::c_int) -> io::Result<c::c_int> {
+    unsafe { ret_c_int(syscall!(__NR_setns, fd, c_int(nstype))) }
```

Comment:
Can this be `syscall_readonly!`, since setns doesn't mutate any memory?

Reply:
> setns doesn't mutate any memory?

`setns` does mutate memory, at least on the kernel level, and maybe even at the level of the process address space, even if that memory is not directly reachable through pointers, but rather through function calls.

Reply:
Yes; `readonly` in this context just refers to the program's own pointers. For example, rustix uses `readonly` with the `write` system call.

Reply:
I didn't find any documentation regarding the `syscall_readonly!()` macro. Can you explain?
1. its purpose,
2. the difference in semantics compared to `syscall!()`, and
3. the consequences of using/misusing it.

Reply:
It's the same as `syscall!` but with the [readonly option](https://doc.rust-lang.org/reference/inline-assembly.html#options) enabled. As documented there, the meaning of readonly in asm is: "does not write to any memory. This allows the compiler to cache the values of unmodified global variables in registers across the asm! block since it knows that they are not written to by the asm!.". Rust isn't aware of memory outside its own address space, so mutating memory in the kernel doesn't impact the readonly option. The consequence for misusing it is Undefined Behavior.

I'll submit a separate PR to add this information in a comment.